src/enclave/Enclave/EnclaveContext.h

```diff
@@ -7,56 +7,88 @@
 #include "../Common/common.h"
 #include "../Common/mCrypto.h"
 
-struct LogEntry;
+struct Crumb;
 
-struct LogEntry {
+struct Crumb {
   int ecall; // ecall executed
-  int snd_pid; // partition where ecall was executed
-  int rcv_pid; // partition of subsequent ecall
-  int job_id; // number of ecalls executed in this enclave before this ecall
+  uint8_t log_mac[OE_HMAC_SIZE]; // LogEntryChain MAC for this output
+  uint8_t all_outputs_mac[OE_HMAC_SIZE]; 
+  // FIXME: change this to num_input_log_macs
+  int num_input_macs; // Num MACS in the below vector
+  std::vector<uint8_t> input_log_macs;
 
-  bool operator==(const LogEntry& le) const
+  bool operator==(const Crumb& c) const
   { 
-      return (this->ecall == le.ecall && this->snd_pid == le.snd_pid && this->rcv_pid == le.rcv_pid && this->job_id == le.job_id); 
+      // Check whether the ecall is the same
+      if (this->ecall != c.ecall) {
+          return false;
+      }
+
+      // Check whether the log_mac and the all_outputs_mac are the same
+      for (int i = 0; i < OE_HMAC_SIZE; i++) {
+          if (this->log_mac[i] != c.log_mac[i]) {
+              return false;
+          }
+          if (this->all_outputs_mac[i] != c.all_outputs_mac[i]) {
+              return false;
+          }
+      }
+      
+      // Check whether input_log_macs size is the same 
+      if (this->input_log_macs.size() != c.input_log_macs.size()) {
+          return false;
+      }
+
+      // Check whether the input_log_macs themselves are the same
+      for (uint32_t i = 0; i < this->input_log_macs.size(); i++) {
+          if (this->input_log_macs[i] != c.input_log_macs[i]) {
+              return false;
+          }
+      }
```

Comment:
I'm a bit confused. Is the `input_log_macs` a sequence of MACs of size `OE_HMAC_SIZE`? If so, does the equality condition also hold when the set of `this->input_log_macs` is the same as that of `c.input_log_macs`?